src/logic/safe/transactions/gas.ts

```diff
@@ -178,94 +180,119 @@ export const getGasEstimationTxResponse = async (txConfig: {
-  fixedGasCosts: number,
-): Promise<number> => {
-  for (const additionalGas of additionalGasBatches) {
-    const batchedSafeTxGas = safeTxGasEstimation + additionalGas
-    // To simulate if safeTxGas is enough we need to send an estimated gasLimit that will be the sum
-    // of the safeTxGasEstimation and fixedGas costs for ethereum transaction
-    const gasLimit = batchedSafeTxGas + fixedGasCosts
-    console.info(`Estimating safeTxGas with gas amount: ${batchedSafeTxGas}`)
-    try {
-      const estimation = await getGasEstimationTxResponse({
-        to: safeAddress,
-        from: safeAddress,
-        data: estimateData,
-        gasPrice: 0,
-        gas: gasLimit,
-      })
-      if (estimation > 0) {
-        console.info(`Gas estimation successfully finished with gas amount: ${batchedSafeTxGas}`)
-        return batchedSafeTxGas
-      }
-    } catch (error) {
-      console.log(`Error trying to estimate gas with amount: ${batchedSafeTxGas}`)
-    }
-  }
-
-  return 0
-}
-
 export const getFixedGasCosts = (threshold: number): number => {
   // There are some minimum gas costs to execute an Ethereum transaction
   // We add this fixed network minimum gas, the gas required to check each signature
   return MINIMUM_TRANSACTION_GAS + (threshold || 1) * GAS_REQUIRED_PER_SIGNATURE
 }
 
-export const estimateGasForTransactionCreation = async (
-  safeAddress: string,
-  data: string,
-  to: string,
-  valueInWei: string,
-  operation: number,
-  safeTxGas?: number,
-): Promise<number> => {
-  try {
-    const safeInstance = await getGnosisSafeInstanceAt(safeAddress)
-
-    const estimateData = safeInstance.methods.requiredTxGas(to, valueInWei, data, operation).encodeABI()
-    const threshold = await safeInstance.methods.getThreshold().call()
-
-    const fixedGasCosts = getFixedGasCosts(Number(threshold))
+type SafeTxGasEstimationProps = {
+  safeAddress: string
+  txData: string
+  txRecipient: string
+  txAmount: string
+  operation: number
+  safeTxGas?: number
+}
 
-    const gasEstimationResponse = await getGasEstimationTxResponse({
-      to: safeAddress,
-      from: safeAddress,
-      data: estimateData,
-      gas: safeTxGas ? safeTxGas + fixedGasCosts : undefined,
+export const estimateSafeTxGas = async ({
+  safeAddress,
+  txData,
+  txRecipient,
+  txAmount,
+  operation,
+  safeTxGas,
+}: SafeTxGasEstimationProps): Promise<number> => {
+  try {
+    const safeTxGasEstimation = await fetchSafeTxGasEstimation({
+      safeAddress,
+      to: checksumAddress(txRecipient),
+      value: txAmount,
+      data: txData,
+      operation,
     })
 
+    console.log('Backend gas estimation', safeTxGasEstimation)
+
     if (safeTxGas) {
-      // When we execute we get a more precise estimate value, we log for debug purposes
-      console.info('This is the smart contract minimum expected safeTxGas', gasEstimationResponse)
+      // If safeTxGas was already defined we leave it but log our estimation for debug purposes
```

Comment:
perhaps, a good thing to do in the future is to check if backend estimation > safeTxGas and somehow add a warning in the UI, because that tx will probably fail.